multimodal/src/autogluon/multimodal/utils/export.py

```diff
@@ -73,82 +76,71 @@ def dump_model(self, save_path: Optional[str] = None):
-        onnx_path: Optional[str] = None,
+        data: pd.DataFrame,
+        path: Optional[str] = None,
         batch_size: Optional[int] = None,
         verbose: Optional[bool] = False,
         opset_version: Optional[int] = 16,
+        truncate_long_and_double: Optional[bool] = False,
     ):
         """
         Export this predictor's model to ONNX file.
 
         Parameters
         ----------
-        onnx_path
-            The export path of onnx model.
         data
             Raw data used to trace and export the model.
             If this is None, will check if a processed batch is provided.
+        path
+            The export path of onnx model.
         batch_size
             The batch_size of export model's input.
             Normally the batch_size is a dynamic axis, so we could use a small value for faster export.
         verbose
             verbose flag in torch.onnx.export.
         opset_version
             opset_version flag in torch.onnx.export.
+        truncate_long_and_double: bool, default False
+            Truncate weights provided in int64 or double (float64) to int32 and float32
+
+        Returns
+        -------
+        trt_module : OnnxModule
+            The onnx-based module that can be used to replace predictor._model for model inference.
         """
-        # TODO: Support CLIP
-        # TODO: Add test
+
         import torch.jit
 
+        from ..models.fusion.fusion_mlp import MultimodalFusionMLP
         from ..models.huggingface_text import HFAutoModelForTextPrediction
         from ..models.timm_image import TimmAutoModelForImagePrediction
 
-        supported_models = (TimmAutoModelForImagePrediction, HFAutoModelForTextPrediction)
+        supported_models = (TimmAutoModelForImagePrediction, HFAutoModelForTextPrediction, MultimodalFusionMLP)
         if not isinstance(self._model, supported_models):
             raise NotImplementedError(f"export_onnx doesn't support model type {type(self._model)}")
         warnings.warn("Currently, the functionality of exporting to ONNX is experimental.")
 
-        valid_input, dynamic_axes, default_onnx_path, batch = get_onnx_input(
-            pipeline=self._problem_type, config=self._config
+        batch = self.get_processed_batch_for_deployment(
+            data=data,
+            onnx_tracing=True,
+            batch_size=batch_size,
         )
 
-        if not batch_size:
-            batch_size = 2  # batch_size should be a dynamic_axis, so we could use a small value for faster export
-        if data is not None:
-            batch = self.get_processed_batch_for_deployment(
-                data=data,
-                valid_input=valid_input,
-                onnx_tracing=True,
-                batch_size=batch_size,
-            )
-
-        if not onnx_path:
-            onnx_path = os.path.join(self.path, default_onnx_path)
-
-        device_type = "cuda" if torch.cuda.is_available() else "cpu"
```

Comment:
Wondering why previously we use GPU for `onnx_export`

Reply:
We actually used cpu instead, see deleted comment below.

To quote
```
# Perform tracing on cpu, since we're facing an error when tracing with cuda device:
#     ERROR: Tensor-valued Constant nodes differed in value across invocations.
#     This often indicates that the tracer has encountered untraceable code.
#     Comparison exception:   The values for attribute 'shape' do not match: torch.Size([]) != torch.Size([384]).
#     from https://github.com/rwightman/pytorch-image-models/blob/3aa31f537d5fbf6be8f1aaf5a36f6bbb4a55a726/timm/models/swin_transformer.py#L112
device = "cpu"
num_gpus = 0
```